src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1484,6 +1484,7 @@ void *CpuemShim::xclMapBO(unsigned int boHandle, bool write)
     if (mLogStream.is_open()) mLogStream << "posix_memalign failed" << std::endl;
     pBuf=nullptr;
   }
+  memset(pBuf, 0, bo->size);
```

Comment:
pBuf can be nullptr per line 1485, this will crash the program

Reply:
Thanks Soren for the feedback, incorporate it and committed the change.

Reply:
Are you sure bo->buf (set on line 1491) will not be garbage when you return without setting it?